### PR TITLE
Move all ha-related tasks under a single block

### PR DESCRIPTION
**Pull request by spetrosi**

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -553,188 +568,221 @@
...
+        - name: Save credentials for the {{ mssql_ha_login }} SQL Server login
+          copy:
+            content: |
+              {{ mssql_ha_login }}
+              {{ mssql_ha_login_password }}
+            dest: /var/opt/mssql/secrets/passwd
+            owner: root
+            group: root
+            mode: 0400
+            force: true
```
Comment (richm) [inline tasks/main.yml:772]: should have a `no_log: true` here

Comment (spetrosi) [reply tasks/main.yml:772]: It should, I should check the role for security everywhere. Add a test with the new vault CI feature

**Comment by spetrosi** [comment]
[citest]

File: tasks/main.yml (hunk 1 above)
Comment (spetrosi) [reply tasks/main.yml:771]: Diff in this PR is very dirty so I will merge it and then continue in future PRs

Comment (spetrosi) [reply tasks/main.yml:771]: Fixing in  #119

**Pull request merged**